controller/planner_backlog_bench_test.go

```diff
@@ -43,69 +39,23 @@ func (rest *BenchPlannerBacklogREST) SetupBenchmark() {
-	application.Transactional(gormapplication.NewGormDB(rest.DB), func(app application.Application) error {
-		spacesRepo := app.Spaces()
-		testSpace = &space.Space{
-			Name: "PlannerBacklogWorkItems-" + uuid.NewV4().String(),
-		}
-		_, err := spacesRepo.Create(rest.Ctx, testSpace)
-		if err != nil {
-			rest.B().Fail()
-		}
-		log.Info(nil, map[string]interface{}{"space_id": testSpace.ID}, "created space")
-		workitemTypesRepo := app.WorkItemTypes()
-		workitemType, err := workitemTypesRepo.Create(rest.Ctx, testSpace.ID, nil, &workitem.SystemPlannerItem, "foo_bar", nil, "fa-bomb", map[string]workitem.FieldDefinition{})
-		if err != nil {
-			rest.B().Fail()
-		}
-		log.Info(nil, map[string]interface{}{"wit_id": workitemType.ID}, "created workitem type")
-
-		iterationsRepo := app.Iterations()
-		parentIteration = &iteration.Iteration{
-			Name:    "Parent Iteration",
-			SpaceID: testSpace.ID,
-			State:   iteration.StateNew,
-		}
-		err = iterationsRepo.Create(rest.Ctx, parentIteration)
-		if err != nil {
-			rest.B().Fail()
-		}
-		log.Info(nil, map[string]interface{}{"parent_iteration_id": parentIteration.ID}, "created parent iteration")
-
-		childIteration := &iteration.Iteration{
-			Name:    "Child Iteration",
-			SpaceID: testSpace.ID,
-			Path:    append(parentIteration.Path, parentIteration.ID),
-			State:   iteration.StateStart,
-		}
-		err = iterationsRepo.Create(rest.Ctx, childIteration)
-		if err != nil {
-			rest.B().Fail()
-		}
-		log.Info(nil, map[string]interface{}{"child_iteration_id": childIteration.ID}, "created child iteration")
-
-		fields := map[string]interface{}{
-			workitem.SystemTitle:     "parentIteration Test",
-			workitem.SystemState:     "new",
-			workitem.SystemIteration: parentIteration.ID.String(),
-		}
-		w, err := app.WorkItems().Create(rest.Ctx, testSpace.ID, workitemType.ID, fields, rest.testIdentity.ID)
-		if w == nil || err != nil {
-			rest.B().Fail()
-		}
-
-		fields2 := map[string]interface{}{
-			workitem.SystemTitle:     "childIteration Test",
-			workitem.SystemState:     "closed",
-			workitem.SystemIteration: childIteration.ID.String(),
-		}
-		createdWI, err = app.WorkItems().Create(rest.Ctx, testSpace.ID, workitemType.ID, fields2, rest.testIdentity.ID)
-		if err != nil {
-			rest.B().Fail()
-		}
-		return nil
-	})
-	return
+	fxt := tf.NewTestFixture(rest.B(), rest.DB,
+		tf.CreateWorkItemEnvironment(),
+		tf.Iterations(2),
+		tf.WorkItems(2, func(fxt *tf.TestFixture, idx int) error {
+			wi := fxt.WorkItems[idx]
+			switch idx {
+			case 0:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateNew
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[0].ID.String()
+			case 1:
+				wi.Fields[workitem.SystemState] = workitem.SystemStateClosed
+				wi.Fields[workitem.SystemIteration] = fxt.Iterations[1].ID.String()
+			}
+			return nil
+		}),
+	)
+	return fxt.Spaces[0], fxt.Iterations[0], fxt.WorkItems[1]
```

Comment:
You create 2 workitems through test fixture. You set their state and iteration. But you return only one workitem. Why?

Reply:
It is a refactoring of the old test setup and I haven't changed that. That's why I only return 1 work item.